python-package/lightgbm/sklearn.py

```diff
@@ -566,6 +571,7 @@ def __init__(
         self._n_features_in: int = -1
         self._classes: Optional[np.ndarray] = None
         self._n_classes: int = -1
+        self.validation_fraction = validation_fraction
```

Comment:
Please move this up to right after `self.importance_type = importance_type`.

If you look carefully, you'll see that all such assignments are ordered by the order of the keyword arguments in the construct... let's please preserve that ordering, to avoid mistakes and make the code easier to read.